src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -0,0 +1,378 @@
+		// Setup values for request timeouts.
+		long ct = longFromStringWithDefault(unirest.get("connectionTimeout"), 10000l);
+		long st = longFromStringWithDefault(unirest.get("socketTimeout"), 10000l);
+		
+		M_log.info("unirest timeouts: connectionTimeout: {} socketTimeout: {}",ct,st);
+		
+		Unirest.setTimeouts(ct,st);
+	}
+
+	// Get a long value from a string use default if the string is null or empty.
+	protected long longFromStringWithDefault(String longString, Long defaultValue) {
+		long longValue = (longString == null ? defaultValue : Long.parseLong(longString));
+		M_log.info("longFromString: {} {}",longString,longValue);
+		return longValue;
+	}
+
+	/*********************** Orchestrate the script. *****************/ 
+	
+	/* 
+	 * - configuration is read and injected by spring
+	 * - get value gradeAfterTime from persist string (default if necessary).
+	 * - get grades.
+	 * - translate to format for updates.
+	 * - send to MPathways.
+	 * 
+	 */
+	
+	/* Organize the task and handle errors */
+	
+	public void orchestrator () throws PersistStringException, SPEEsbException {
+	
+		M_log.info("Start SPE orchestrator");
+	
+		setUnirestGlobalValues();
+		
+		// sanity check by renewing token to ensure we can connect to the database.
+		
+		if (!verifyESB()) {
+			M_log.error("Unable to connect to esb"); 
+			throw new SPEEsbException("Unable to connect to ESB");
+		}
+		
+		//// Get the relevant grades.
+		String lastUpdateTime = readLastGradeTransferTime();
+		M_log.error("fake lastUpdateTime");
+		lastUpdateTime = "2017-03-08 15:38:21";
+		M_log.info("lastUpdateTime: {}",lastUpdateTime);
+		String assignmentsFromDW;
+	
+		try {
+			assignmentsFromDW = getSPEGrades(lastUpdateTime);
+	
+			//// Extract grades from JSON the grades for insertion	
+	
+			ArrayList<HashMap<String,String>> SPEgradeMaps = null;
+			try {
+				SPEgradeMaps = convertSPEGradesFromDataWarehouseJSON(assignmentsFromDW);
+			} catch (JSONException e) {
+				M_log.error("exception in converting grades: "+e);
+			}
+	
+			//// Insert the grades.
+			M_log.info("Grade count since {} is {}.",lastUpdateTime,SPEgradeMaps.size());
+			putSPEGrades(SPEgradeMaps);
+	
+		} catch (SPEEsbException e1) {
+			M_log.error("Exception processing SPE grades:",e1);
+		}
+		finally {
+			// Finish up, save data, send reports.
+			closeUpShop();
+		}
+	}
+
+	/************ Close down processing, create summary ************/
+	public void closeUpShop() {
+		// TODO: implement task shutdown.
+		M_log.error("Implement close up shop");
+	}
+
+	/********* esb verify **********/
+	// Run sanity check to make sure can talk to ESB and get data.
+	
+	/*
+	 * Setup the hash of values needed for the call to verify ESB connection.
+	 */
+	protected HashMap<String, String> setupESBVerifyCall() {
+		//M_log.debug("spe properties: "+speproperties);
+		HashMap<String,String> value = new HashMap<String,String>();
+		value.putAll(speproperties.getEsb());
+		return value;
+	}
+
+	public Boolean verifyESB() {
```

Comment:
I think here it's not verifying ESB but verifying if we can connect to ESB. A better name would be `VerifyESBConnection` or something like that